src/audio/module_adapter/module/volume/volume.c

```diff
@@ -263,84 +259,83 @@ static int32_t volume_windows_fade_ramp(struct processing_module *mod, int32_t r
 
 /**
  * \brief Ramps volume changes over time.
- * \param[in,out] mod Volume processing module handle
-
+ * \param[in,out] vol_data Volume component data
  */
-static void volume_ramp(struct processing_module *mod)
+static inline void volume_ramp(struct vol_data *cd)
```

Comment:
hm, interesting, I'd literary expect the compiler to do that for us automatically. Since the gain isn't huge, to confirm that there's really a difference - could you perhaps just build the firmware with and without `inline` and check that the resulting image changed (in size)? At least `volume_windows_fade_ramp()` really shouldn't need `inline` since you take its address:
```
	cd->ramp_func =  &volume_windows_fade_ramp;
```

Reply:
ok, so we dont have time to figure out why we get the XCC speed improvement with inline here, lets just mark this with an inline comment so that its obvious.